lib/handler/connect.c

```diff
@@ -174,8 +197,8 @@ static int on_req(h2o_handler_t *_handler, h2o_req_t *req)
     h2o_timer_link(creq->loop, handler->config.tunnel.timeout, &creq->timeout);
 
     creq->getaddr_req = h2o_hostinfo_getaddr(&creq->src_req->conn->ctx->receivers.hostinfo_getaddr, creq->server_name.host,
-                                             h2o_iovec_init(creq->server_name.port, port_strlen), AF_UNSPEC, SOCK_STREAM,
-                                             IPPROTO_TCP, AI_ADDRCONFIG | AI_NUMERICSERV, on_getaddr, creq);
+            h2o_iovec_init(creq->server_name.port, port_strlen), AF_UNSPEC, SOCK_STREAM,
+            IPPROTO_TCP, AI_ADDRCONFIG | AI_NUMERICSERV, on_getaddr, creq);
```

Comment:
I believe that SOCK_STREAM and IPPROTO_TCP should be something else for UDP. And, assuming that these values can be obtained from one of the arguments passed to `on_getaddr` callback, we can get rid of `connect_type`.